quic/s2n-quic-transport/src/endpoint/mod.rs

```diff
@@ -1051,6 +1063,33 @@ impl<Cfg: Config> Endpoint<Cfg> {
         let limits = endpoint_context
             .connection_limits
             .on_connection(&LimitsInfo::new(&remote_address));
+
+        let mut endpoint_publisher = event::EndpointPublisherSubscriber::new(
+            event::builder::EndpointMeta {
+                endpoint_type: Cfg::ENDPOINT_TYPE,
+                timestamp,
+            },
+            Some(quic_version),
+            endpoint_context.event_subscriber,
+        );
+        let info = mtu::ConnectionInfo::new(&remote_address, self.endpoint_mtu_config);
+        let mtu_config = CheckedConfig::new(endpoint_context.mtu.on_connection(&info), &info)
+            .map_err(|_err| {
+                let error =
+                    connection::Error::validation("failed to instantiate a valid MTU provider");
```

Comment:
I wouldn't say you weren't able to instantiate a valid MTU provider, its that the MTU provider failed to produce a valid MTU configuration